tests/cases/arithmetic_decimal/power.test

```diff
@@ -0,0 +1,21 @@
+### SUBSTRAIT_SCALAR_TEST: 1.0
+### SUBSTRAIT_INCLUDE: extensions/functions_arithmetic_decimal.yaml
+
+# basic: Basic examples without any special cases
+power(8::decimal, 2::decimal<38, 0>) = 64::fp64
+power(1.0::decimal, -1.0::decimal<38, 0>) = 1.0::fp64
+power(2.0::decimal<38, 0>, -2.0::decimal<38, 0>) = 0.25::fp64
+power(13::decimal<38, 0>, 10::decimal<38, 0>) = 137858491849::fp64
+
+# result_more_than_input_precison: Examples demonstrating result with more precision than input
+power(16::decimal<2, 0>, 4::decimal<38, 0>) = 65536::fp64
```

Comment:
Perhaps we could reuse the text format's definition of literals?

Reply:
My preference would be to stick with :: format generally. It is a cast format consistent with many systems the people looking at these tests will be familiar with and I believe much more intuitive understood than underscore.

I'm similarly inclined to try to find more standard ways to represent other things as opposed to what the text format has done. There are several examples of alternative ways to implement things to what is commonly done in SQL (which has the most consistent way to approach these things across systems). Examples beyond the use of cast syntax versus underscores include. (1) String literals are double quoted and backticked while sql string literals are single quoted and (2) brackets being overloaded to mean any of intervals, lists, maps and structs.

Reply:
The :: format is a type subclass operator.  As such it requires a mental context switch every time I look at it.  The underscore was chosen in the text format to create a visual separation between the literal value and its type thus aiding in readability (and it's also requires less keystrokes).

Braces are used for literal values and angle brackets are used for type composition and parameterization.  This helps to reinforce the distinction between value and type.  I have yet to be able to determine what the value of any interval literal is in the proposed format.  However the text format's use of labels makes it ultra-clear what each part of the interval is.

The text format's string implementation does draw from other sources than SQL but that's mostly because the SQL version has its own warts which make it harder to read.  SQL's implicit concatenation is the main SQL string feature that is difficult for humans to parse (it requires the counting of active single quotes).


The text format was reviewed in the community meeting twice (February 1st, 2023 and April 12th, 2023) so presumably there is some community agreement about its design validity.

Reply:
It's not that the text format defined in substrait-cpp isn't valid. It's definitely valid. 

I think we come from different perspectives. I'm coming from the perspective that we should try to leverage existing patterns as much as possible. My expectation is that the vast majority of systems that will be tested are SQL systems so leveraging SQL patterns is desirable as much as reasonably possible and comes with the added benefit of being obvious in meaning to a large group of people reviewing the test case. If we look at the set of BFT dialects, for example, we have the following:

| Dialect    | API       | Single Quotes for Literals | Supports :: for Cast |
|------------|-----------|----------------------------|----------------------|
| cudf       | cudf api  | no                         | no                   |
| datafusion | SQL       | yes                        | no                   |
| duckdb     | SQL       | yes                        | yes                  |
| postgres   | SQL       | yes                        | yes                  |
| snowflake  | SQL       | yes                        | yes                  |
| sqllite    | SQL       | yes                        | no                   |
| velox      | velox api | no                         | no                   |


So a decent portion are SQL based. Single-quoted literals are the most common way to represent strings in those systems. 3/7 have support for the double colon as meaning cast. Given that, I prefer that over something invented.